src/io/flutter/actions/DeviceSelectorAction.java

```diff
@@ -20,27 +32,36 @@
 import java.util.Objects;
 import java.util.stream.Collectors;
 
-
 public class DeviceSelectorAction extends ComboBoxAction implements DumbAware {
-
   final private List<SelectDeviceAction> actions = new ArrayList<>();
+  private boolean isListening = false;
 
   @NotNull
   @Override
   protected DefaultActionGroup createPopupActionGroup(JComponent button) {
     final DefaultActionGroup group = new DefaultActionGroup();
-    FlutterDaemonService service = FlutterDaemonService.getInstance();
-    if (service != null) {
-      final Collection<ConnectedDevice> devices = service.getConnectedDevices();
-      actions.clear();
-      actions.addAll(devices.stream().map(SelectDeviceAction::new).collect(Collectors.toList()));
-      if (actions.isEmpty()) {
-        group.add(new NoDevicesAction());
-      }
-      else {
-        group.addAll(actions);
+
+    updateActions();
+
+    if (actions.isEmpty()) {
+      group.add(new NoDevicesAction());
+    }
+    else {
+      group.addAll(actions);
+    }
+
+    if (SystemInfo.isMac) {
+      boolean simulatorOpen = false;
```

Comment:
We pass it into the `OpenSimulatorAction` ctor, int order to toggle enablement of the action.